src/com/epam/javatrainings/Validator.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.javatrainings;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.regex.*;
+import java.time.Period;
+
+public class Validator <T> {
+    public static final Pattern VALID_EMAIL_ADDRESS_REGEX =  Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+    private ArrayList<Error> errorList;
+    private T dto;
+
+    public Validator(T dto){
+        this.dto = dto;
+        errorList = new ArrayList<Error>();
+    }
+
+    public ArrayList <Error> Validate() {
+        try {
+            Class cls = dto.getClass();
+            Field nameField = cls.getDeclaredField("name");
+            Length nameAnnotation = nameField.getAnnotation(Length.class);
+            Method getName = cls.getMethod("getName");
+            String name = (String) getName.invoke(dto);
+            if (name.length() < nameAnnotation.min()) {
+                errorList.add(new Error("The length of " + nameField.getName() + " must be at least " + nameAnnotation.min() + '.'));
+            }
+            if (name.length() > nameAnnotation.max()) {
```

Comment:
may produce NullPointerException if @Length annotation is not presenting on field